unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+import uuid
+from unittest import mock
+
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+from unittest.mock import MagicMock
+from unittest.mock import Mock
+from unittest.mock import patch
+
+area_id = str(uuid.uuid4())
+area_uid = None
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.publish = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.pubsub.psubscribe = MagicMock()
+
+
+@pytest.fixture
+def redis_client_auto_register():
+    return RedisClientBase(area_id=area_id, autoregister=False)
+
+
+class TestRedisClientBase:
+    @classmethod
+    def setup_class(cls):
+        cls.obj = RedisClientBase(area_id=area_id, autoregister=False)
+
+    def test_constructor_registry_success(self, pubsub_thread=None):
+        assert self.obj.area_id == area_id
+        assert self.obj.area_uuid == area_uid
+        assert self.obj.is_active == False
+        assert self.obj._subscribed_aggregator_response_cb is None
+        assert self.obj._subscribe_to_response_channels(pubsub_thread) is None
+        assert self.obj._blocking_command_responses == {}
+        assert self.obj._transaction_id_buffer == []
+
+    def test_function_subscribe_to_response_channels_success(self):
+        RedisClientBase._subscribe_to_response_channels = MagicMock()
```

Comment:
Nit: in theory (following my comment above about testing private methods) we wouldn't need to assert that this method was called. In this specific case you might need it for other reasons, but I encourage you to give it a look and see if it is necessary.

Reply:
Another comment on this: why are you mocking directly the `RedisClientBase` class, instead of the `redis_client_auto_register` fixture?